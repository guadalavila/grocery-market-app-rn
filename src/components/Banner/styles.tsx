import { StyleSheet } from 'react-native';
import { Colors } from '../../utils/Colors';
import { Sizes } from '../../utils/GlobalStyles';

export const styles = StyleSheet.create({
    container: {
        marginHorizontal: Sizes.M,
        backgroundColor: Colors.primaryLight,
        marginVertical: Sizes.M,
        borderRadius: Sizes.L,
        paddingHorizontal: Sizes.M,
        height: 120,
    },
    image: {
        height: 100,
        position: 'absolute',
        left: 10,
        top: 10,
    },
    smalText: {
        marginTop: Sizes.L,
        fontSize: 20,
        color: Colors.green,
        fontWeight: '500',
        marginLeft: Sizes.L,
    },
    bigText: {
        fontSize: 38,
        color: Colors.green,
        fontWeight: '900',
        textAlign: 'center',
    },
});
