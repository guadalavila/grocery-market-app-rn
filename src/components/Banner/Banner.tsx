import React from 'react';
import { Image, Text, View } from 'react-native';
import { styles } from './styles';

interface IBannerProps {
    smallText: string;
    bigText: string;
}

const Banner = ({ smallText, bigText }: IBannerProps) => {
    return (
        <View style={[styles.container]}>
            <Text style={styles.smalText}>{smallText}</Text>
            <Text style={styles.bigText}>{bigText}</Text>
            <Image
                testID='image'
                style={styles.image}
                resizeMode='center'
                source={require('../../../assets/images/carrot.png')}
            />
        </View>
    );
};

export default Banner;
