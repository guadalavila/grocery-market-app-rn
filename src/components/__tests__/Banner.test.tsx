import React from 'react';
import { render, screen } from '@testing-library/react-native';
import Banner from '../Banner/Banner';

describe('Banner', () => {
    const firstText = 'I am small Text';
    const secondText = 'I am big text';

    test('should match snapshot ', () => {
        const banner = render(<Banner smallText={firstText} bigText={secondText} />);
        expect(banner).toMatchSnapshot();
    });

    test('renders the correct message', () => {
        render(<Banner smallText={firstText} bigText={secondText} />);
        expect(screen.getByText(firstText)).toBeTruthy();
        expect(screen.getByText(secondText)).toBeTruthy();
    });

    test('should render image', () => {
        render(<Banner smallText={firstText} bigText={secondText} />);
        expect(screen.getByTestId('image')).toBeTruthy();
        // screen.debug();
    });
});
