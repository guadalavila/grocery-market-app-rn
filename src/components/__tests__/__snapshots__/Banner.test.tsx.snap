// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should match snapshot  1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#CFFDE1",
        "borderRadius": 24,
        "height": 120,
        "marginHorizontal": 16,
        "marginVertical": 16,
        "paddingHorizontal": 16,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#2DA32A",
        "fontSize": 20,
        "fontWeight": "500",
        "marginLeft": 24,
        "marginTop": 24,
      }
    }
  >
    I am small Text
  </Text>
  <Text
    style={
      Object {
        "color": "#2DA32A",
        "fontSize": 38,
        "fontWeight": "900",
        "textAlign": "center",
      }
    }
  >
    I am big text
  </Text>
  <Image
    resizeMode="center"
    source={
      Object {
        "testUri": "../../../assets/images/carrot.png",
      }
    }
    style={
      Object {
        "height": 100,
        "left": 10,
        "position": "absolute",
        "top": 10,
      }
    }
    testID="image"
  />
</View>
`;
